src/main/kotlin/solve/main/MainController.kt

```diff
@@ -4,13 +4,22 @@ import solve.catalogue.controller.CatalogueController
 import solve.project.model.ProjectFrame
 import solve.scene.SceneFacade
 import solve.scene.controller.SceneController
-import solve.utils.ServiceLocator
+import solve.utils.*
 import tornadofx.*
 
 class MainController : Controller() {
+    private val view: MainView by inject()
```

Comment:
It is not MVC approach, add observable property and do actions when it changes

Reply:
In this case, the controller gives an interface for communicating different modules with each other